google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/models/QueryTest.java

```diff
@@ -286,4 +286,29 @@ public void testEquality() {
         .isNotEqualTo(Query.create(TABLE_ID).filter(FILTERS.family().exactMatch("test-one")));
     assertThat(Query.create(TABLE_ID).limit(4)).isNotEqualTo(Query.create(TABLE_ID).limit(5));
   }
+
+  @Test
+  public void testClone() {
+    Query query = Query.create(TABLE_ID).filter(FILTERS.key().regex("temp")).limit(10);
+    ReadRowsRequest request =
+        ReadRowsRequest.newBuilder()
+            .setTableName(NameUtil.formatTableName(PROJECT_ID, INSTANCE_ID, TABLE_ID))
+            .setAppProfileId(APP_PROFILE_ID)
+            .setRowsLimit(10)
+            .setFilter(
+                RowFilter.newBuilder()
+                    .setRowKeyRegexFilter(ByteString.copyFromUtf8("temp"))
+                    .build())
+            .build();
+
+    assertThat(query.toProto(requestContext)).isEqualTo(request);
+
+    assertThat(Query.create(TABLE_ID).limit(5).toProto(requestContext))
+        .isNotEqualTo(
+            ReadRowsRequest.newBuilder()
+                .setTableName(NameUtil.formatTableName(PROJECT_ID, INSTANCE_ID, TABLE_ID))
+                .setAppProfileId(APP_PROFILE_ID)
+                .setRowsLimit(10)
+                .build());
```

Comment:
What are you testing here?

Reply:
Thanks for pointing it out, Have updated this with valid test case.